internal/service/gateway/connection.go

```diff
@@ -72,6 +138,24 @@ func ResourceConnection() *schema.Resource {
 	}
 }
 
+func resourceConnectionStateUpgradeV0(ctx context.Context, rawState map[string]any, meta any) (map[string]any, error) {
+	svc := meta.(*service.Service)
+	conns, err := svc.GetGatewayConnections(ctx, &request.GetGatewayConnectionsRequest{ServiceUUID: rawState["gateway"].(string)})
+	if err != nil {
+		return rawState, err
+	}
+
+	for _, conn := range conns {
+		if conn.Name == rawState["name"].(string) {
+			rawState["uuid"] = conn.UUID
+
+			return rawState, nil
+		}
+	}
+
+	return rawState, fmt.Errorf("connection by name %s not found", rawState["name"].(string))
+}
+
 func resourceConnectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
```

Comment:
Should we use the UUID of the created resource in the `d.SetId(...)`? (Could not add comment to unchanged line 176)

Reply:
Changed to use the UUID, better for import purposes.